packages/contracts/src/core/dao/DAO.sol

```diff
@@ -122,6 +154,21 @@ contract DAO is
         __PermissionManager_init(_initialOwner);
     }
 
+    /// @notice Initializes the DAO after an upgrade from a previous version.
+    /// @param previousProtocolVersion The protocol version number of the previous DAO implementation contract this upgrade is transitioning from.
+    function initializeUpgradeFrom(
+        uint8[3] calldata previousProtocolVersion
+    ) external reinitializer(2) {
+        if (previousProtocolVersion[0] != 1 || previousProtocolVersion[1] >= 3) {
+            revert ProtocolVersionUpgradeNotSupported(previousProtocolVersion);
+        }
```

Comment:
According to what we have published so far, is this really possible to hit this "if"? If we can save gas I would do it

Reply:
See here my answer here: https://github.com/aragon/osx/pull/355/files/f689e6ce2d0ab95f29343a192dd0d13b6d5a789b#r1192063403